pkg/ccl/changefeedccl/sink_webhook.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package changefeedccl
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/httputil"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/errors"
+)
+
+const (
+	applicationTypeJSON = `application/json`
+	authorizationHeader = `Authorization`
+)
+
+func isWebhookSink(u *url.URL) bool {
+	switch u.Scheme {
+	// allow HTTP here but throw an error later to make it clear HTTPS is required
+	case changefeedbase.SinkSchemeWebhookHTTP, changefeedbase.SinkSchemeWebhookHTTPS:
+		return true
+	default:
+		return false
+	}
+}
+
+type webhookSinkPayload struct {
+	Payload []json.RawMessage `json:"payload"`
+}
+
+func encodePayloadWebhook(value []byte) ([]byte, error) {
+	var payload json.RawMessage
+	if err := json.Unmarshal(value, &payload); err != nil {
+		return nil, err
+	}
```

Comment:
I think we should be able to avoid this Unmarshal now that we are using json.RawMessage by casting the []byte to a json.RawMessage.